File: 백준/DFS와BFS.py
from collections import deque

N, M, V = map(int, input().split())

graph = {}

for _ in range(M):
    l, r = map(int, input().split())
    if l not in graph:
        graph[l] = [r]
    else:
        graph[l].append(r)
    
    if r not in graph:
        graph[r] = [l]
    else:
        graph[r].append(l)

for i in graph.keys():
    graph[i].sort()



visited = [0] * N
path = []
def dfs(v):
    if v not in graph:
        return
    
    for i in graph[v]:
        if (visited[i - 1] == 0):
            path.append(i)
            visited[i - 1] = 1
            dfs(i)

path.append(V)
visited[V - 1] = 1
dfs(V)
print(*path)


visited = [0] * N
path = []
def bfs():
    queue = deque()
    queue.append(V)

    while queue:
        v = queue.popleft()
        if visited[v - 1] == 0:
            path.append(v)
            visited[v - 1] = 1
            if v in graph:
                for i in graph[v]:
                    queue.append(i)
bfs()
print(*path)

File: 백준/DFS와BFS.cpp
#include <iostream>
#include <vector>
#include <queue>
#include <algorithm>

using namespace std;

vector<int> adj[1001];
bool visited[1001];


void dfs(int v){
    visited[v] = true;
    cout << v << " ";
    for (int next : adj[v]) {
        if (!visited[next]) dfs(next);
    }
}

void bfs(int start) {
    queue<int> q;
    q.push(start);
    visited[start] = true;

    while(!q.empty()) {
        int v = q.front();
        q.pop();
        cout << v << " ";
        for (int next : adj[v]) {
            if (!visited[next]) {
                visited[next] = true;
                q.push(next);
            }
        }
    }
}


int main() {
    int N, M, V;

    cin >> N >> M >> V;

    for (int i = 0; i < M; i++) {
        int u, v;
        cin >> u >> v;
        adj[u].push_back(v);
        adj[v].push_back(u);
    }

    for(int i = 1; i <= N; i++) {
        sort(adj[i].begin(), adj[i].end());
    }

    dfs(V);
    cout << "\n";

    for(int i = 1; i <= N; i++) visited[i] = false;

    bfs(V);
    cout << "\n";

    return 0;
}